design/aws-ecs-autoscaling-queue-worker.md

```diff
@@ -0,0 +1,225 @@
+# AWS ECS - L3 Construct for Autoscaling ECS/Fargate Service that Processes Items in a SQS Queue
+
+To address issue [#2396](https://github.com/awslabs/aws-cdk/issues/2396), the AWS ECS CDK construct library should provide a way for customers to create a Queue Worker service (an AWS ECS/Fargate Service that processes items from an sqs queue). This construct will create an SQS Queue and enable autoscaling for the service based on cpu usage and the sqs approximateNumberOfMessagesVisible metric. This would mean adding new ECS CDK constructs `Ec2QueueWorkerService` and `FargateQueryWorkerService`, that would take in the necessary properties required to create a Task Definition, an SQS Queue as well as an ECS/Fargate Service with Auto Scaling rules.
+
+## General approach
+
+The new [`ecs.QueueWorkerServiceBase`], [`ecs.Ec2QueueWorkerService`] and [`ecs.FargateQueueWorkerService`] classes will include L3 constructs for:
+
+* Ec2QueueWorkerService
+* FargateQueueWorkerService
```

Comment:
This is confusing wording -- you're stating that the constructs that are listed include L3 constructs? Perhaps it would be clearer if what was listed instead was the underlying L2 constructs that we're building on top of?

Reply:
Oh, I do see you do that in line 95. Might make more sense to move that up here then?

Reply:
Updated